pkg/openstack/client/client.go

```diff
@@ -46,10 +50,27 @@ func NewOpenstackClientFromCredentials(credentials *os.Credentials) (Factory, er
 		},
 	}
 
+	config := &tls.Config{
+		InsecureSkipVerify: credentials.Insecure,
+	}
+	if len(credentials.CACert) > 0 {
+		pool := x509.NewCertPool()
+		ok := pool.AppendCertsFromPEM([]byte(credentials.CACert))
+		if !ok {
+			return nil, fmt.Errorf("failed to load CA Bundle for KeyStone")
+		}
+		config.RootCAs = pool
+	}
+
 	if opts.AuthInfo.ApplicationCredentialSecret != "" {
 		opts.AuthType = clientconfig.AuthV3ApplicationCredential
 	}
 
+	transport := &http.Transport{Proxy: http.ProxyFromEnvironment, TLSClientConfig: config}
+	opts.HTTPClient = &http.Client{
+		Transport: transport,
+	}
+
```

Comment:
@kon-angelo It seems like settings opts.HTTPClient doesn't have any effect, since it isn't passed through to `authOpts`. Have you actually seen this work in action? 
Our feeling is that https://github.com/gardener/gardener-extension-provider-openstack/commit/72a3c33c79dd4bb5ae16fdbe6a9f243fe6ec8c99 might already be enough to tackle the problem.

@lotharbach @jensac

Reply:
I think that is reasonable. It is what gophercloud is also doing in it's utils package [ref](https://github.com/gophercloud/utils/blob/de873b9cf00d4b57e384f3f8e6a52dfcb290ecd8/openstack/clientconfig/requests.go#L826-L833). 

Thank you for pointing it that out @j2L4e. Can you open a PR for it ?